db/blob/db_blob_basic_test.cc

```diff
@@ -210,6 +210,36 @@ TEST_F(DBBlobBasicTest, GetBlob_IndexWithInvalidFileNumber) {
                   .IsCorruption());
 }
 
+#ifndef ROCKSDB_LITE
+TEST_F(DBBlobBasicTest, GenerateIOTracing) {
+  Options options = GetDefaultOptions();
+  options.enable_blob_files = true;
+  options.min_blob_size = 0;
+
+  Reopen(options);
+
+  std::string trace_file = dbname_ + "/io_trace_file";
+  std::unique_ptr<TraceWriter> trace_writer;
+  ASSERT_OK(NewFileTraceWriter(env_, EnvOptions(), trace_file, &trace_writer));
+
+  ASSERT_OK(db_->StartIOTrace(env_, TraceOptions(), std::move(trace_writer)));
+
+  constexpr char key[] = "key";
+  constexpr char blob_value[] = "blob_value";
+
+  ASSERT_OK(Put(key, blob_value));
+  ASSERT_OK(Flush());
+  ASSERT_EQ(Get(key), blob_value);
+
+  ASSERT_OK(db_->EndIOTrace());
+  ASSERT_OK(env_->FileExists(trace_file));
```

Comment:
Question: is there a way to make sure that blob file I/Os specifically got traced, using e.g. `IOTraceReader`? I believe this test case would pass even without the current changes b/c we also read/write an SST.